### PR TITLE
Update cloudflare_utils.py

### PR DESCRIPTION
**Pull request by amalmurali47**
Added a function for getting the subnets directly from the website.

### CODE REVIEW

**Comment by christophetd** [comment]
Hi, 

Thanks a lot for the PR! I'll review it ASAP, but overall it looks good. 👍 

Cheers

File: cloudflare_utils.py [hunk 1]
```diff
@@ -1,43 +1,42 @@
...
-    "173.245.48.0/20",
-    "188.114.96.0/20",
-    "190.93.240.0/20",
-    "197.234.240.0/22",
-    "198.41.128.0/17"
-]
+import requests
+
+
+def get_cloudflare_ip_ranges():
```
Comment (christophetd) [inline cloudflare_utils.py]: Would be great to have a bit more logging, e.g. `[*] Retrieving Cloudflare IP ranges from https://www.cloudflare.com/ips-v4`

File: cloudflare_utils.py [hunk 2]
```diff
@@ -1,43 +1,42 @@
...
+    cloudflare_ip_ranges_url = 'https://www.cloudflare.com/ips-v4'
+    try:
+        page_content = requests.get(cloudflare_ip_ranges_url, timeout=10)
+        ip_ranges_text = page_content.text
+        return [ip for ip in ip_ranges_text.split("\n") if ip]
+    except requests.exceptions.RequestException:
+        pass
```
Comment (christophetd) [inline cloudflare_utils.py]: If we fail to retrieve IP ranges from Cloudflare, how about:

- Using the current hardcoded list
- Printing an error message on stderr `[-] Failed to retrieve Cloudflare IP ranges - using a default (possibly outdated) list`

?

File: cloudflare_utils.py [hunk 3]
```diff
@@ -1,43 +1,42 @@
...
+
+
+cloudflare_ip_ranges = get_cloudflare_ip_ranges()
 
 if sys.version_info[0] == 2:
-    cloudflare_subnets = [ ipaddress.ip_network(ip_range.decode('utf-8')) for ip_range in cloudflare_ip_ranges ]
+    cloudflare_subnets = [ipaddress.ip_network(ip_range.decode('utf-8')) for ip_range in cloudflare_ip_ranges]
```
Comment (christophetd) [inline cloudflare_utils.py]: Any reason why you removed the spaces after / before the square brackets? Is this the commonly recommended Python style?

File: cloudflare_utils.py (hunk 2 above)
Comment (amalmurali47) [reply cloudflare_utils.py]: Sounds like a better idea.

File: cloudflare_utils.py (hunk 3 above)
Comment (amalmurali47) [reply cloudflare_utils.py]: Yes. PEP8 standards: http://pep8.org/

File: cloudflare_utils.py (hunk 1 above)
Comment (amalmurali47) [reply cloudflare_utils.py]: That is true. Informing the user of the current action is better.

File: cloudflare_utils.py (hunk 3 above)
Comment (christophetd) [reply cloudflare_utils.py]: Good to know, thanks.

**Comment by amalmurali47** [comment]
@christophetd I've made those changes. Please review.

File: cloudflare_utils.py [hunk 4]
```diff
@@ -9,12 +9,36 @@
...
     try:
+        print('[*] Retrieving Cloudflare IP ranges from https://www.cloudflare.com/ips-v4')
         page_content = requests.get(cloudflare_ip_ranges_url, timeout=10)
         ip_ranges_text = page_content.text
-        return [ip for ip in ip_ranges_text.split("\n") if ip]
+        ip_ranges = [ip for ip in ip_ranges_text.split("\n") if ip]
+
     except requests.exceptions.RequestException:
-        pass
+        sys.stderr.write('Failed to retrieve Cloudflare IP ranges - using a default (possibly outdated) list')
```
Comment (christophetd) [inline cloudflare_utils.py]: You're missing a `\n` at the end (which `sys.stderr.write` doesn't append)

Comment (christophetd) [reply cloudflare_utils.py]: And maybe add a `[-]` before the error message, so we more easily see it's an error :)

File: cloudflare_utils.py [hunk 5]
```diff
@@ -9,12 +9,36 @@
...
+        "172.64.0.0/13",
+        "173.245.48.0/20",
+        "188.114.96.0/20",
+        "190.93.240.0/20",
+        "197.234.240.0/22",
+        "198.41.128.0/17"
+    ]
+
     try:
+        print('[*] Retrieving Cloudflare IP ranges from https://www.cloudflare.com/ips-v4')
```
Comment (christophetd) [inline cloudflare_utils.py]: Maybe reuse `cloudflare_ip_ranges_url` instead of hardcoding it?

File: cloudflare_utils.py (hunk 4 above)
Comment (amalmurali47) [reply cloudflare_utils.py]: Ah, you're right. Fixed.

**Pull request merged**